From 58f7d6e86c6d9face706598fb36d80ce90f0e27f Mon Sep 17 00:00:00 2001
From: Eyal Reizer <eyalr@ti.com>
Date: Wed, 6 Jun 2012 19:07:43 +0530
Subject: [PATCH 1007/1092] ARM: OMAP2+: am335x: Add wlan suspend-resume
 support

Configure the wlan_enable pin to output mode on every resume
as its configuration is lost during suspend

MMC_PM_KEEP_POWER flag is removed from the .caps structure element
since it needs to a part of .pm_caps, if required.

Signed-off-by: Eyal Reizer <eyalr@ti.com>
[vaibhav.bedia@ti.com - Update the patch to take care of the
recent changes in the board file, drop the pad context save
and restore since that is now handled in pm33xx.c and update
the commit message accordingly]
Signed-off-by: Vaibhav Bedia <vaibhav.bedia@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index aea9617..f47b27b 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -1513,8 +1513,7 @@ static void mmc2_wl12xx_init(int evm_id, int profile)
 
 	am335x_mmc[1].mmc = 3;
 	am335x_mmc[1].name = "wl1271";
-	am335x_mmc[1].caps = MMC_CAP_4_BIT_DATA | MMC_CAP_POWER_OFF_CARD
-				| MMC_PM_KEEP_POWER;
+	am335x_mmc[1].caps = MMC_CAP_4_BIT_DATA | MMC_CAP_POWER_OFF_CARD;
 	am335x_mmc[1].nonremovable = true;
 	am335x_mmc[1].gpio_cd = -EINVAL;
 	am335x_mmc[1].gpio_wp = -EINVAL;
@@ -1543,11 +1542,11 @@ static void wl12xx_bluetooth_enable(void)
 static int wl12xx_set_power(struct device *dev, int slot, int on, int vdd)
 {
 	if (on) {
-		gpio_set_value(am335xevm_wlan_data.wlan_enable_gpio, 1);
+		gpio_direction_output(am335xevm_wlan_data.wlan_enable_gpio, 1);
 		mdelay(70);
+	} else {
+		gpio_direction_output(am335xevm_wlan_data.wlan_enable_gpio, 0);
 	}
-	else
-		gpio_set_value(am335xevm_wlan_data.wlan_enable_gpio, 0);
 
 	return 0;
 }
-- 
1.7.11.2

